torch/_inductor/index_propagation.py

```diff
@@ -345,4 +346,12 @@ def wrap_expr(expr):
                     dict(lower=not can_prove_lower, upper=not can_prove_upper),
                 )
             return expr
-        return self.fallback("indirect_indexing", (index, size, check), {}).value
+
+        indirect_var = self.fallback(
+            "indirect_indexing", (index, size, check), {}
+        ).value
+        if indirect_var not in self.var_to_range:
+            lower, upper = -upper_bound(size), upper_bound(size) - 1
+            indirect_range = (indirect_var, ValueRanges(lower, upper))
+            self.var_to_range = self.var_to_range + (indirect_range,)
```

Comment:
Can this be turned into
```suggestion
        assert indirect_var not in self.var_to_range
        lower, upper = -upper_bound(size), upper_bound(size) - 1
        indirect_range = (indirect_var, ValueRanges(lower, upper))
        self.var_to_range = self.var_to_range + (indirect_range,)
```
?

Reply:
sure!